### PR TITLE
Feature/erssup 60546

### PR DESCRIPTION
**Pull request by jamiegreen4**
## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: proxies/live/apiproxy/targets/ers-target.xml [hunk 1]
```diff
@@ -89,19 +89,38 @@
...
         <Flow name="app-restricted-flow">
             <Condition>(accesstoken.auth_type == "app")</Condition>
-            <Request>
-                <Step>
-                    <Name>RaiseFault.403Forbidden</Name>
-                </Step>
-            </Request>
+            <Request><!-- reject if headers that are overriden as part of app rectricted call are provided--><Step>
+                    <Name>RaiseFault.400BadRequest</Name>
+                    <Condition>(request.header.x-ers-ods-code != null)</Condition>
```
Comment (andrewclark2020) [inline proxies/live/apiproxy/targets/ers-target.xml:101]: Do you want != null or do you want to just detect whether the header is present or not? If you just want to check for the header then remove the != null

File: proxies/live/apiproxy/targets/ers-target.xml [hunk 2]
```diff
@@ -89,19 +89,38 @@
...
+                </Step> <Step>
+                    <Name>RaiseFault.400BadRequest</Name>
+                    <Condition>(request.header.x-ers-business-function != null)</Condition>
```
Comment (andrewclark2020) [inline proxies/live/apiproxy/targets/ers-target.xml:104]: Same here... remove != null if you just want to check for presence of it

File: proxies/live/apiproxy/targets/ers-target.xml [hunk 3]
```diff
@@ -89,19 +89,38 @@
...
+                </Step> <Step>
+                    <Name>RaiseFault.400BadRequest</Name>
+                    <Condition>(request.header.x-ers-user-id != null)</Condition>
```
Comment (andrewclark2020) [inline proxies/live/apiproxy/targets/ers-target.xml:107]: Same here... remove != null if you just want to check for presence of it

File: proxies/live/apiproxy/targets/ers-target.xml [hunk 4]
```diff
@@ -89,19 +89,38 @@
...
+                    <Name>AssignMessage.Set.nhsd-ers-ods-code-header-app-restricted</Name>
+                </Step> <Step>
+                    <Name>AssignMessage.Set.nhsd-ers-business-function-header-app-restricted</Name>
+                </Step> <Step>
+                    <Name>AssignMessage.Set.x-ers-user-id-header-app-restricted</Name>
+                </Step><Step>
+                    <Name>AssignMessage.Remove.x-request-id-header</Name>
+                </Step> {% if ALLOW_ECHO_TARGET | default(false) == true %} <Step>
+                    <Name>AssignMessage.SetEchoTarget</Name>
+                    <Condition>(request.header.echo)</Condition>
```
Comment (andrewclark2020) [inline proxies/live/apiproxy/targets/ers-target.xml:118]: As per this line here which checks for the presence of the header not caring about the value

File: proxies/live/apiproxy/targets/ers-target.xml (hunk 1 above)
Comment (jamiegreen4) [reply proxies/live/apiproxy/targets/ers-target.xml:101]: Must admit I got this from a much earlier branch. I'll remove the unnecessary null check

File: proxies/live/apiproxy/targets/ers-target.xml (hunk 2 above)
Comment (jamiegreen4) [reply proxies/live/apiproxy/targets/ers-target.xml]: Will remove

File: proxies/live/apiproxy/targets/ers-target.xml (hunk 3 above)
Comment (jamiegreen4) [reply proxies/live/apiproxy/targets/ers-target.xml]: Will remove

File: proxies/live/apiproxy/policies/AssignMessage.Remove.x-ers-business-function-header.xml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
```
Comment (andrewclark2020) [inline proxies/live/apiproxy/policies/AssignMessage.Remove.x-ers-business-function-header.xml:1]: Are all these policy changes around headers tested in the Test_Headers test?

**Review by nhsd-jack-wainwright — COMMENTED** [review]
I've only had quite a brief look at the changes included here but they look fine to me 🙂

File: proxies/live/apiproxy/policies/AssignMessage.Remove.x-ers-business-function-header.xml (hunk 5 above)
Comment (jamiegreen4) [reply proxies/live/apiproxy/policies/AssignMessage.Remove.x-ers-business-function-header.xml]: I've added a test to cover this

**Review by andrewclark2020 — COMMENTED** [review]
GitHub makes it difficult to see what's changed since the last time I reviewed it so only focussed on the echo target tests which look fine. I'm going to presume you've now fully tested all the rules around headers in the new test.

**Review by nhsd-jack-wainwright — APPROVED** [review]
Looks good to me though I've only had a brief look over the changes here. I believe these changes will have already been reviewed previously though 🙂

**Pull request merged**